keras/ops/image.py

```diff
@@ -501,3 +501,157 @@ def map_coordinates(
+            self.offset_width,
+            self.target_height,
+            self.target_width,
+            self.check_dims,
+        )
+
+    def compute_output_spec(self, image):
+        out_shape = (
+            image.shape[0],
+            self.target_height,
+            self.target_width,
+            image.shape[-1],
+        )
+        if len(image.shape) == 3:
+            out_shape = out_shape[1:]
+        return KerasTensor(
+            shape=out_shape,
+            dtype=image.dtype,
+        )
+
+
+def _pad_to_bounding_box(
+    image, offset_height, offset_width, target_height, target_width, check_dims
+):
+    image = backend.convert_to_tensor(image)
+    is_batch = True
+    image_shape = image.shape
+    if len(image_shape) == 3:
+        is_batch = False
+        image = backend.numpy.expand_dims(image, 0)
+    elif len(image_shape) != 4:
+        raise ValueError(
+            "'image' (shape %s) must have either 3 or 4 dimensions."
+            % image_shape
+        )
+
+    batch, height, width, depth = image.shape
+    after_padding_width = target_width - offset_width - width
+    after_padding_height = target_height - offset_height - height
+
+    if check_dims:
+        if not offset_height >= 0:
+            raise ValueError("offset_height must be >= 0")
+        if not offset_width >= 0:
+            raise ValueError("offset_width must be >= 0")
+        if not after_padding_width >= 0:
+            raise ValueError("width must be <= target - offset")
+        if not after_padding_height >= 0:
+            raise ValueError("height must be <= target - offset")
+
+    paddings = backend.numpy.reshape(
+        backend.numpy.stack(
+            [
+                0,
+                0,
+                offset_height,
+                after_padding_height,
+                offset_width,
+                after_padding_width,
+                0,
+                0,
+            ]
+        ),
+        [4, 2],
+    )
+    padded = backend.numpy.pad(image, paddings)
+
+    padded_shape = [batch, target_height, target_width, depth]
+    padded = backend.numpy.reshape(padded, padded_shape)
+
+    if not is_batch:
+        padded = backend.numpy.squeeze(padded, axis=[0])
+    return padded
+
+
+@keras_export("keras.ops.image.pad_to_bounding_box")
+def pad_to_bounding_box(
+    image,
+    offset_height,
+    offset_width,
+    target_height,
+    target_width,
+    check_dims=True,
+):
+    """Pad `image` with zeros to the specified `height` and `width`.
+
+    Args:
+        image: 4-D Tensor of shape `[batch, height, width, channels]` or 3-D
+            Tensor of shape `[height, width, channels]`.
+        offset_height: Number of rows of zeros to add on top.
+        offset_width: Number of columns of zeros to add on the left.
+        target_height: Height of output image.
+        target_width: Width of output image.
+        check_dims: If True, assert that dimensions are non-negative and in
```

Comment:
We should not need this argument. If shapes are static, we can raise the error in every case (at creation time). If shapes are dynamic (negative dims are not detectable until runtime), let's not assert anything.

Reply:
I believe in `tf.image.pad_to_bounding_box`, they determine whether to use static or dynamic dimensions by checking `ndims` in `image.get_shape()` [here](https://github.com/tensorflow/tensorflow/blob/4dacf3f368eb7965e9b5c3bbdd5193986081c3b2/tensorflow/python/ops/image_ops_impl.py#L1116). However, I am unsure how to implement this for multibackend. Any suggestions would be greatly appreciated.